allennlp/tests/data/vocabulary_test.py

```diff
@@ -671,3 +671,17 @@ def test_max_vocab_size_dict(self):
         words = vocab.get_index_to_token_vocabulary().values()
         # Additional 2 tokens are '@@PADDING@@' and '@@UNKNOWN@@' by default
         assert len(words) == 3
+
+    def test_max_vocab_size_partial_dict(self):
+        indexers = {"tokens": SingleIdTokenIndexer(), "token_characters": TokenCharactersIndexer()}
+        instance = Instance({
+            'text': TextField([Token(w) for w in 'The quick brown fox lay down for a rest'.split(' ')], indexers)
+        })
+        dataset = Batch([instance])
+        params = Params({
+                "max_vocab_size": {
+                        "tokens": 1
+                }
+        })
+
+        vocab = Vocabulary.from_params(params=params, instances=dataset)
```

Comment:
It'd be better to actually have an assert here, that checks that the token vocabulary size is 1, and the token characters vocabulary size is 20-something (or whatever it actually is).